src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -62,15 +62,22 @@ private static IEnumerable<ICompilationRootProvider> GetCompilationRoots(IEnumer
 
         protected override void CompileInternal(string outputFile, ObjectDumper dumper)
         {
-            using (FileStream inputFile = File.OpenRead(_inputFilePath))
+            foreach (var inputFilePath in _inputFilePaths)
             {
-                PEReader inputPeReader = new PEReader(inputFile);
+                using (FileStream inputFile = File.OpenRead(inputFilePath))
+                {
+                    PEReader inputPeReader = new PEReader(inputFile);
 
-                _dependencyGraph.ComputeMarkedNodes();
-                var nodes = _dependencyGraph.MarkedNodeList;
+                    _dependencyGraph.ComputeMarkedNodes();
+                    var nodes = _dependencyGraph.MarkedNodeList;
```

Comment:
This will not work: NodeFactory and dependency graph that was used by the prior compilation cannot be reused. You need to get new ones. What you can reuse is the type system context (it will already be "warmed up" by the previous compilation and a lot of the information will already be cached).

The correct flow for this would be to create and configure several `ReadyToRunCompilationBuilders` in Program.cs and doing individual compilations, with a shared compiler type system context.

(Or just invoke ilc.exe several times in parallel from the .targets files - *not from ilc.exe itself*.)